core/src/main/java/io/fabric8/maven/core/service/openshift/OpenshiftBuildService.java

```diff
@@ -374,31 +374,30 @@ private Build startBuild(OpenShiftClient client, File dockerTar, String buildNam
 
     private void waitForOpenShiftBuildToComplete(OpenShiftClient client, Build build) throws MojoExecutionException, InterruptedException {
         final CountDownLatch latch = new CountDownLatch(1);
+        final CountDownLatch readyLatch = new CountDownLatch(1);
         final CountDownLatch logTerminateLatch = new CountDownLatch(1);
         final String buildName = KubernetesHelper.getName(build);
-
-        final AtomicReference<Build> buildHolder = new AtomicReference<>();
+        final String buildPodName = buildName + "-build";
 
         // Don't query for logs directly, Watch over the build pod:
-        waitUntilPodIsReady(buildName + "-build", 20, log);
-        log.info("Waiting for build " + buildName + " to complete...");
-        try (LogWatch logWatch = client.pods().withName(buildName + "-build").watchLog()) {
-            KubernetesClientUtil.printLogsAsync(logWatch,
-                    "Failed to tail build log", logTerminateLatch, log);
-            Watcher<Build> buildWatcher = getBuildWatcher(latch, buildName, buildHolder);
-            try (Watch watcher = client.builds().withName(buildName).watch(buildWatcher)) {
+        Watcher<Pod> buildWatcher = getBuildWatcher(latch, readyLatch, buildPodName);
+        try (Watch watcher = client.pods().withName(buildPodName).watch(buildWatcher)) {
+            log.info("Waiting for build " + buildName + " to complete...");
+            waitForBuildEvent(readyLatch);
```

Comment:
This blocks the thread until the latch is released.

Reply:
Yes, the intention is to wait until pods gets into `Ready` state so that logs can be queried into it. Since Openshift 3.7 additional pod states:
```
fabric8-maven-sample-spring-boot-s2i-1-build   0/1       Pending   0         0s
fabric8-maven-sample-spring-boot-s2i-1-build   0/1       Init:0/2   0         0s
fabric8-maven-sample-spring-boot-s2i-1-build   0/1       Init:0/2   0         2s
fabric8-maven-sample-spring-boot-s2i-1-build   0/1       Init:1/2   0         3s
fabric8-maven-sample-spring-boot-s2i-1-build   0/1       PodInitializing   0         4s
fabric8-maven-sample-spring-boot-s2i-1-build   1/1       Running   0         5s

```